source/common/http/conn_manager_utility.cc

```diff
@@ -176,14 +176,49 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
     request_headers.insertRequestId().value(uuid);
   }
 
-  if (config.tracingConfig()) {
-    Tracing::HttpTracerUtility::mutateHeaders(request_headers, runtime);
-  }
+  mutateTracingRequestHeader(request_headers, runtime, config);
   mutateXfccRequestHeader(request_headers, connection, config);
 
   return final_remote_address;
 }
 
+void ConnectionManagerUtility::mutateTracingRequestHeader(Http::HeaderMap& request_headers,
+                                                          Runtime::Loader& runtime,
+                                                          ConnectionManagerConfig& config) {
+  if (!config.tracingConfig() || !request_headers.RequestId()) {
+    return;
+  }
+
+  std::string x_request_id = request_headers.RequestId()->value().c_str();
+  uint64_t result;
+  // Skip if x-request-id is corrupted.
+  if (!UuidUtils::uuidModBy(x_request_id, result, 10000)) {
+    return;
+  }
+
+  // Do not apply tracing transformations if we are currently tracing.
+  if (UuidTraceStatus::NoTrace == UuidUtils::isTraceableUuid(x_request_id)) {
+    if (request_headers.ClientTraceId() &&
+        (runtime.snapshot().featureEnabled("tracing.client_enabled",
```

Comment:
What is the default if this is not set?

Reply:
Default is 100

Reply:
where do you set that default to be 100? Does the proto handle that?

Reply:
Yes, the proto accepts Percent which is converted to 100 uint for client_sampling and overall_samping and 10000 for random sampling. Kept the default same as the runtime values but am thinking should we make all the sampling parameters more flexible by allowing setting of values like 0.001?